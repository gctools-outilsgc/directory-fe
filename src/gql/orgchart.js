import gql from 'graphql-tag';

export const GET = gql`
query getOrgChart($gcIDa: String!, $gcIDb: String, $leftGutter: Float) {
  orgchart(gcIDa: $gcIDa, gcIDb: $gcIDb, leftGutter: $leftGutter) {
    boxes {
      id
      x
      y
      on_path
      width
      height
      node {
        avatar
        name
        titleEn
        root
        x
        y
      }
    }
    lines {
      id
      d
      on_path
    }
    miniboxes {
      id
      x
      y
      on_path
      width
      height
      node {
        avatar
        name
        titleEn
        root
        x
        y
      }
    }
    minilines {
      id
      d
      on_path
    }
  }
}`;

export default GET;
