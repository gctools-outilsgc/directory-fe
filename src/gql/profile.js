import gql from 'graphql-tag';

export const GET = gql`
query getProfile($gcID: String!) {
  profiles(gcID: $gcID) {
    gcID
    name
    email
    avatar
    mobilePhone
    officePhone
    address {
      id
      streetAddress
      city
      province
      postalCode
      country
    }
    titleEn
    titleFr
  }
}`;

export const GET_TEAM = gql`
query getTeam($gcID: String!) {
  profiles(gcID: $gcID) {
    gcID
    supervisor {
      gcID
      name
      titleEn
      titleFr
    }
    team {
      nameEn
      nameFr
    }
  }
}`;

const profileDataForOrgChart = gql`
  fragment profileDataForOrgChart on Profile {
    gcID
    name
    avatar
    titleEn
    titleFr
  }
`;

const teamDataForOrgChart = gql`
  fragment teamDataForOrgChart on Team {
    id
    nameEn
    nameFr
  }
`;

export const ORGCHART = gql`
query orgChart($gcID: String!) {
  profiles(gcID: $gcID) {
    ...profileDataForOrgChart
    ownerOfTeams {
      ...teamDataForOrgChart
      members {
        ...profileDataForOrgChart
      }
    }
    team {
      ...teamDataForOrgChart
      owner {
        ...profileDataForOrgChart
      }
      members {
        ...profileDataForOrgChart
      }
    }
  }
}
${profileDataForOrgChart}
${teamDataForOrgChart}
`;


export const EDIT = gql`
mutation editProfile($gcID: String!, $data: ModifyProfileInput!) {
  modifyProfile(gcID: $gcID, data: $data) {
    gcID
    name
    email
    avatar
    mobilePhone
    officePhone
    address {
      id
      streetAddress
      city
      province
      postalCode
      country
    }
    titleEn
    titleFr
    supervisor {
      gcID
    }
  }
}
`;

/**
 * Prepare variables suitable for the editProfile mutation
 * @param {object} data Object with the following keys:
 * @param {string} data.gcID Profile's GCID
 * @param {string} data.name Name (required)
 * @param {string} data.email Email (required)
 * @param {string} data.titleEn English title
 * @param {string} data.titleFr French title
 * @param {string} data.officePhone Office phone number
 * @param {string} data.mobilePhone Mobile phone
 * @param {string} data.streetAddress Street Address
 * @param {string} data.city City
 * @param {string} data.province Province
 * @param {string} data.postalCode Postal code
 * @param {string} data.country Country
 */
export const prepareEditProfile = (data) => {
  const valueOrUndefined = (v) => {
    if (typeof v === 'object') {
      const test = Object.keys(v).reduce((t, c) =>
        t || typeof v[c] !== 'undefined', false);
      return (test) ? v : undefined;
    }
    if (v && v !== '') return v;
    return undefined;
  };
  const {
    gcID,
    name, email, titleEn, titleFr, officePhone, mobilePhone,
    streetAddress, city, province, postalCode, country,
  } = data;
  return {
    variables: {
      gcID,
      data: {
        name,
        email,
        titleEn: valueOrUndefined(titleEn),
        titleFr: valueOrUndefined(titleFr),
        officePhone: valueOrUndefined(officePhone),
        mobilePhone: valueOrUndefined(mobilePhone),
        address: valueOrUndefined({
          streetAddress: valueOrUndefined(streetAddress),
          city: valueOrUndefined(city),
          province: valueOrUndefined(province),
          postalCode: valueOrUndefined(postalCode),
          country: valueOrUndefined(country),
        }),
      },
    },
  };
};

export default GET;
