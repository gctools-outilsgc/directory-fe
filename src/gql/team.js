import gql from 'graphql-tag';

export const CREATE_TEAM = gql`
mutation createTeam(
  $nameEn: String!,
  $nameFr: String!,
  $descriptionEn: String,
  $descriptionFr: String,
  $organization: OrganizationInput!,
  $owner: TeamOwnerInput!
)
{
  createTeam(
    nameEn: $nameEn,
    nameFr: $nameFr,
    descriptionEn: $descriptionEn,
    descriptionFr: $descriptionFr,
    organization: $organization,
    owner: $owner
  ) {
    id
    nameEn
    nameFr
    descriptionEn
    descriptionFr
    organization {
      id
    }
    owner {
      gcID
      name
    }
  }
}`;

export default CREATE_TEAM;
