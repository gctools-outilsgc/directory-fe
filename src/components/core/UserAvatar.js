import React from 'react';
import PropTypes from 'prop-types';

import { Mutation } from 'react-apollo';
import { Button } from 'reactstrap';

import styled from 'styled-components';

import { EDIT } from '../../gql/profile';
import GenericAvatar from '../profile/OrgChart/Card/img/user.gif';


const UploadContainer = styled.div`
  text-align: center;
  margin-top: 10px;
`;

export const UserAvatar = (props) => {
  const {
    myGcID,
    edit,
    name,
    avatar: profileAvatar,
    gcID,
  } = props;
  const canEdit = edit && (gcID === myGcID);
  const avatar = profileAvatar || GenericAvatar;

  return (
    <div>
      <div className="query">
        <div>
          <img
            className="avatar avatar-lg rounded-circle"
            src={avatar}
            alt={name}
          />
        </div>
        {canEdit && (
        <UploadContainer className="mutate">
          <Mutation
            mutation={EDIT}
          >
            {uploadAvatar => (
              <Button>
                <label htmlFor="avatarUploadTest">
                  Upload
                  <input
                    type="file"
                    id="avatarUploadTest"
                    style={{ display: 'none' }}
                    required
                    onChange={({ target }) => {
                      uploadAvatar({
                        variables: {
                          gcID: myGcID,
                          data: {
                            avatar: target.files[0],
                          },
                        },
                      });
                    }}
                  />
                </label>
              </Button>
            )}
          </Mutation>
        </UploadContainer>
        )}
      </div>
    </div>
  );
};

UserAvatar.defaultProps = {
  edit: false,
  name: 'No photo',
  gcID: '',
  myGcID: '',
  avatar: '',
};

UserAvatar.propTypes = {
  edit: PropTypes.bool,
  gcID: PropTypes.string,
  name: PropTypes.string,
  avatar: PropTypes.string,
  myGcID: PropTypes.string,
};

export default UserAvatar;

