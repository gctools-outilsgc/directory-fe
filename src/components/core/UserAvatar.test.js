import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import { MockedProvider } from 'react-apollo/test-utils';

import { UserAvatar } from './UserAvatar';
import { EDIT } from '../../gql/profile';

const mockProps = {
  edit: true,
  gcID: '2',
  name: 'First Last',
  avatar: '',
  myGcID: '2',
};

const mockProvider = [
  {
    request: {
      query: EDIT,
      variables: {
        gcID: mockProps.gcID,
        data: {
          avatar: 'http://test.com/avatar',
        },
      },
    },
    result: {
      data: {
        avatar: 'http://test.com/avatar',
      },
    },
  },
];

describe('UserAvatar', () => {
  it('renders the component', () => {
    const component =
      render(<MockedProvider><UserAvatar {...mockProps} /></MockedProvider>);
    expect(component).toBeTruthy();
  });
  it('uploading photo sets the photo', () => {
    const { getByText, asFragment } =
      render((
        <MockedProvider mocks={mockProvider}>
          <UserAvatar {...mockProps} />
        </MockedProvider>
      ));
    const firstRender = asFragment();
    fireEvent.click(getByText(/Upload/));
    expect(firstRender).toMatchSnapshot(asFragment());
  });
});
