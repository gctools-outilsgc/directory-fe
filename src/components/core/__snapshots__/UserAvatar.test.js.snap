// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar uploading photo sets the photo 1`] = `Object {}`;
